### ISSUE TITLE
System.Speech is missing API documentation

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#87711]
See https://github.com/dotnet/runtime/pull/84917 for more context.

The compiler generated XML file for System.Speech is shipping to customers via the nuget package and is missing public API documentation. The above linked PR disables the compiler error CS1591 until all the API is documented.

When working on this, remove the NoWarn=1591 setting from the project file.

**Comment by stephentoub** [comment]
I'm surprised we're shipping the compiler-generated XML file rather than one derived from the docs.

**Comment by ViktorHofer** [comment]
> I'm surprised we're shipping the compiler-generated XML file rather than one derived from the docs.

The docs package (`Microsoft.Private.Intellisense.nupkg`) doesn't provide an XML file for System.Speech, nor does it for a couple of other assemblies today:
- https://github.com/dotnet/runtime/issues/87709
- https://github.com/dotnet/runtime/issues/87710

Instead of getting a documentation file added to the package (which presumably would be work for the docs team), we prefer to have the source-of-truth in runtime: https://github.com/dotnet/runtime/issues/44969. @carlossanlop and Akhil were working on porting tool that should ease that process.

**Comment by ViktorHofer** [comment]
FWIW these are all the current docs source-of-truth assemblies in runtime (please ignore all the facade assemblies):
- Microsoft.Bcl.AsyncInterfaces
- Microsoft.Bcl.Cryptography
- Microsoft.Bcl.TimeProvider
- Microsoft.Extensions.DependencyInjection.Specification.Tests
- Microsoft.Extensions.Diagnostics
- Microsoft.Extensions.Diagnostics.Abstractions
- Microsoft.Extensions.Hosting.Systemd
- Microsoft.Extensions.Hosting.WindowsServices
- Microsoft.NET.WebAssembly.Webcil
- Microsoft.VisualBasic
- System
- System.AppContext
- System.Buffers
- System.ComponentModel.DataAnnotations
- System.Configuration
- System.Core
- System.Data
- System.Data.DataSetExtensions
- System.Diagnostics.Debug
- System.Diagnostics.EventLog.Messages
- System.Diagnostics.Tools
- System.Drawing
- System.Dynamic.Runtime
- System.Formats.Cbor
- System.Globalization
- System.Globalization.Calendars
- System.Globalization.Extensions
- System.IO
- System.IO.Compression.FileSystem
- System.IO.FileSystem
- System.IO.FileSystem.Primitives
- System.IO.UnmanagedMemoryStream
- System.Net
- System.Numerics
- System.Numerics.Tensors
- System.Reflection
- System.Reflection.Extensions
- System.Resources.Reader
- System.Resources.ResourceManager
- System.Runtime.CompilerServices.Unsafe
- System.Runtime.Extensions
- System.Runtime.Handles
- System.Runtime.InteropServices.RuntimeInformation
- System.Runtime.Serialization
- System.Runtime.Serialization.Schema
- System.Security
- System.Security.Cryptography.Algorithms
- System.Security.Cryptography.Cng
- System.Security.Cryptography.Csp
- System.Security.Cryptography.Encoding
- System.Security.Cryptography.OpenSsl
- System.Security.Cryptography.Primitives
- System.Security.Cryptography.X509Certificates
- System.Security.Principal
- System.Security.SecureString
- System.ServiceModel.Web
- System.ServiceProcess
- System.Speech
- System.Text.Encoding
- System.Threading.RateLimiting
- System.Threading.Tasks
- System.Threading.Tasks.Extensions
- System.Threading.Timer
- System.Transactions
- System.ValueTuple
- System.Web
- System.Windows
- System.Xml
- System.Xml.Linq
- System.Xml.Serialization
- System.Xml.XmlDocument
- WindowsBase
- Microsoft.XmlSerializer.Generator
- mscorlib
- netstandard

**Comment by stephentoub** [comment]
> we prefer to have the source-of-truth in runtime

Yes, but it's not for Speech, as is evidenced by this issue and the facts that the docs are more complete than the xml comments.

**Comment by ericstj** [comment]
This is happening because the `Microsoft.Private.Intellisense` package is missing `System.Speech`.  @carlossanlop any idea how we can get that added?  Alternatively we can just run porting tool on the .NETFramework XMLs to get it ported to source.
[System.Speech.zip](https://github.com/dotnet/runtime/files/12285572/System.Speech.zip)

**Comment by ericstj** [comment]
@carlossanlop - can you see about getting this added to the intellisense package?

**Comment by carlossanlop** [comment]
Step 1: Finish documenting all APIs in System.Speech in dotnet-api-docs. Only 4 APIs were incomplete. I have a PR for that: https://github.com/dotnet/dotnet-api-docs/pull/9336

Step 2: Understand why there's no System.Speech assembly in the ref assemblies drop that we generate for dotnet-api-docs. The item exists in [Versions.props](https://github.com/dotnet/api-catalog-infra/blob/4472bd7a87002896b09e0fb6fb45c92dbef89d84/eng/Versions.props#L92) and [Versions.Details.xml](https://github.com/dotnet/api-catalog-infra/blob/4472bd7a87002896b09e0fb6fb45c92dbef89d84/eng/Version.Details.xml#L643)  of api-catalog-infra, but the assembly file is not getting published in the ref assemblies drop when we build, so there are no docs for that assembly included in the [the Microsoft.Private.Intellisense package](https://dnceng.visualstudio.com/public/_artifacts/feed/dotnet8-transport/NuGet/Microsoft.Private.Intellisense/versions/8.0.0-preview-20230918.1).

Step 3: Once figured out the root cause, send an updated ref-assemblies drop to dotnet-api-docs, then generate a new Microsoft.Private.Intellisense version, confirm it contains an xml for System.Speech, and finally confirm it works for customers in intellisense.

Step 4 (Future): backport all the docs of System.Spech from dotnet-api-docs using the [PortToTripleSlash tool](https://github.com/dotnet/api-docs-sync) and re-enable the property in the csproj so that triple slash docs become the source of truth for this assembly.

**Comment by ViktorHofer** [comment]
> Step 2: Understand why there's no System.Speech assembly in the ref assemblies drop that we generate for dotnet-api-docs. The item exists in [Versions.props](https://github.com/dotnet/api-catalog-infra/blob/4472bd7a87002896b09e0fb6fb45c92dbef89d84/eng/Versions.props#L92) and [Versions.Details.xml](https://github.com/dotnet/api-catalog-infra/blob/4472bd7a87002896b09e0fb6fb45c92dbef89d84/eng/Version.Details.xml#L643) of api-catalog-infra, but the assembly file is not getting published in the ref assemblies drop when we build, so there are no docs for that assembly included in the [the Microsoft.Private.Intellisense package](https://dnceng.visualstudio.com/public/_artifacts/feed/dotnet8-transport/NuGet/Microsoft.Private.Intellisense/versions/8.0.0-preview-20230918.1).

The reference assembly is part of the `Microsoft.DotNet.Runtime.Docs.Transport` package. Shouldn't we use that instead for runtime and aspnetcore (those are the repos that already publish a transport docs package).

<img width="518" alt="image" src="https://github.com/dotnet/runtime/assets/7412651/99e2aac0-8d0d-4f7c-b858-c05c5edb9d24">

**Comment by ericstj** [comment]
Just chiming in that I think this is the reason for step 2: https://github.com/dotnet/api-catalog-infra/pull/73
Agree with @ViktorHofer that if we created new transport packages to eliminate some of the complexity in api-catalog-infra using that seems preferrable.

Also, for Step 4, we're still blocked because this assembly has a RID-specific impl with PNSE.  I considered doing porting the docs to fix this package but did not for this reason.

**Comment by gewarren** [comment]
> Step 4 (Future): When the PNSE problem is fixed, backport all the docs of System.Speech from dotnet-api-docs using the [PortToTripleSlash tool](https://github.com/dotnet/api-docs-sync) and re-enable the property in the csproj so that triple slash docs become the source of truth for this assembly.

I'm a bit confused here. This project does not explicitly [set UseCompilerGeneratedDocXmlFile to false](https://github.com/dotnet/runtime/blob/c5bb5057dddf4b5834626e90e63bfd50f0cda3ff/src/libraries/System.IO.Hashing/src/System.IO.Hashing.csproj#L6), meaning that the triple slash docs are already considered the source of truth for this assembly.

Also, I no longer use the transport packages when updating dotnet-api-docs, so Step 2 is no longer a problem. (For this particular assembly, I use the System.Numerics.Tensors NuGet package directly.)
